File: bildunya_frontend/lib/data/models/conversation_summary_dto.dart
/// [com.bildunya.dto.ConversationSummaryDto]
class ConversationSummaryDto {
  const ConversationSummaryDto({
    this.id,
    this.otherUserId,
    this.otherUsername,
    this.otherFullName,
    this.lastMessage,
    this.lastMessageAt,
    this.unreadCount,
    this.relatedContentId,
    this.relatedContentLabel,
  });

  final int? id;
  final int? otherUserId;
  final String? otherUsername;
  final String? otherFullName;
  final String? lastMessage;
  final String? lastMessageAt;
  final int? unreadCount;
  final int? relatedContentId;
  final String? relatedContentLabel;

  factory ConversationSummaryDto.fromJson(Map<String, dynamic> json) {
    return ConversationSummaryDto(
      id: (json['id'] as num?)?.toInt(),
      otherUserId: (json['other_user_id'] as num?)?.toInt() ??
          (json['otherUserId'] as num?)?.toInt(),
      otherUsername: json['other_username'] as String? ??
          json['otherUsername'] as String?,
      otherFullName: json['other_full_name'] as String? ??
          json['otherFullName'] as String?,
      lastMessage:
          json['last_message'] as String? ?? json['lastMessage'] as String?,
      lastMessageAt: json['last_message_at'] as String? ??
          json['lastMessageAt'] as String?,
      unreadCount: (json['unread_count'] as num?)?.toInt() ??
          (json['unreadCount'] as num?)?.toInt(),
      relatedContentId: (json['related_content_id'] as num?)?.toInt() ??
          (json['relatedContentId'] as num?)?.toInt(),
      relatedContentLabel: json['related_content_label'] as String? ??
          json['relatedContentLabel'] as String?,
    );
  }
}

File: BilDunya/src/main/java/com/bildunya/dto/ConversationSummaryDto.java
package com.bildunya.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ConversationSummaryDto {

    private Long id;

    @JsonProperty("other_user_id")
    private Long otherUserId;

    @JsonProperty("other_username")
    private String otherUsername;

    @JsonProperty("other_full_name")
    private String otherFullName;

    @JsonProperty("last_message")
    private String lastMessage;

    @JsonProperty("last_message_at")
    private String lastMessageAt;

    @JsonProperty("unread_count")
    private Long unreadCount;

    @JsonProperty("related_content_id")
    private Long relatedContentId;

    @JsonProperty("related_content_label")
    private String relatedContentLabel;
}

